src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,141 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) values (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
```

Comment:
Need changes here, try with resources is better to used only for read operation with DB

Reply:
++, thank you, changed